src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -389,5 +378,44 @@ internal void RemoveStream(long streamId)
                 _streams.Remove(streamId);
             }
         }
+
+        public bool SetInboundControlStream(Http3ControlStream stream)
+        {
+            lock (_sync)
```

Comment:
Is 
```c#
public bool SetInboundControlStream(Http3ControlStream stream)
    => Interlocked.CompareExchange(ref _inboundControlStream, stream, null) == null;
```
enough or must it be locked on `_sync`?

Reply:
I had that originally (first commit), but I don't quite believe it's enough. I believe if an encoding stream is just starting to be processed, which sets the control stream, and the stream is being aborted at the same time, there could be a race.